test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -81,10 +113,11 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
                 applicationBasePath,
                 applicationWebSiteName);
             services.AddInstance<IApplicationEnvironment>(environment);
+            services.AddInstance<IHostingEnvironment>(new HostingEnvironment(environment));
```

Comment:
from the way you've updated the functional tests, it appears this is the only required change to this class.  plus likely the `using` updates.  revert adding the new, unused `CreateServer`, the `private` method renaming, and the minor change below.